src/app/tests/TestReportScheduler.cpp

```diff
@@ -352,28 +350,29 @@ class TestReportScheduler
 
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler1->SetMaxReportingInterval(2));
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler1->SetMinReportingIntervalForTests(1));
-        ReadHandlerNode * node = sScheduler.FindReadHandlerNode(readHandler1);
-        node->SetIntervalTimeStamps(readHandler1);
-        readHandler1->MoveToState(ReadHandler::HandlerState::GeneratingReports);
+        sScheduler.OnReadHandlerSubscribed(readHandler1);
+        readHandler1->ClearStateFlag(ReadHandler::ReadHandlerFlags::PrimingReports);
+        readHandler1->MoveToState(ReadHandler::HandlerState::CanStartReporting);
         readHandler1->ForceDirtyState();
 
         // Clean read handler, will be triggered at max interval
         ReadHandler * readHandler2 =
             readHandlerPool.CreateObject(nullCallback, exchangeCtx, ReadHandler::InteractionType::Subscribe, &sScheduler);
+
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler2->SetMaxReportingInterval(3));
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler2->SetMinReportingIntervalForTests(0));
-        node = sScheduler.FindReadHandlerNode(readHandler2);
-        node->SetIntervalTimeStamps(readHandler2);
-        readHandler2->MoveToState(ReadHandler::HandlerState::GeneratingReports);
+        sScheduler.OnReadHandlerSubscribed(readHandler2);
+        readHandler2->ClearStateFlag(ReadHandler::ReadHandlerFlags::PrimingReports);
+        readHandler2->MoveToState(ReadHandler::HandlerState::CanStartReporting);
 
         // Clean read handler, will be triggered at max interval, but will be cancelled before
         ReadHandler * readHandler3 =
             readHandlerPool.CreateObject(nullCallback, exchangeCtx, ReadHandler::InteractionType::Subscribe, &sScheduler);
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler3->SetMaxReportingInterval(3));
         NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == readHandler3->SetMinReportingIntervalForTests(0));
-        node = sScheduler.FindReadHandlerNode(readHandler3);
-        node->SetIntervalTimeStamps(readHandler3);
-        readHandler3->MoveToState(ReadHandler::HandlerState::GeneratingReports);
+        sScheduler.OnReadHandlerSubscribed(readHandler3);
+        readHandler3->ClearStateFlag(ReadHandler::ReadHandlerFlags::PrimingReports);
```

Comment:
Again, this should match the "real" ordering, so we are testing things closer to what will actually run.